src/game/models/mod.rs

```diff
@@ -36,6 +36,16 @@ pub trait Positioned {
     }
 
     fn pos(&self) -> vector::Vector;
+
+    fn dx(&self) -> f64 {
```

Comment:
Also, is there any benefit to using [trait inheritance ](https://doc.rust-lang.org/book/traits.html#inheritance) to leave `Positioned` as a trait, but then have a derived trait for the "moving" types?